File: app/src/main/java/com/mdmobile/cyclops/dataTypes/ApiActions.java
package com.mdmobile.cyclops.dataTypes;

import android.support.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import static com.mdmobile.cyclops.dataTypes.ApiActions.CHECKIN;
import static com.mdmobile.cyclops.dataTypes.ApiActions.LOCATE;
import static com.mdmobile.cyclops.dataTypes.ApiActions.LOCK;
import static com.mdmobile.cyclops.dataTypes.ApiActions.SEND_MESSAGE;
import static com.mdmobile.cyclops.dataTypes.ApiActions.SEND_SCRIPT;
import static com.mdmobile.cyclops.dataTypes.ApiActions.UNENROL;
import static com.mdmobile.cyclops.dataTypes.ApiActions.WIPE;

/**
 * Constants to define possible api actions
 */
@SuppressWarnings("unused")
@StringDef({CHECKIN, WIPE, LOCK, UNENROL, SEND_MESSAGE, LOCATE, SEND_SCRIPT})
@Retention(RetentionPolicy.SOURCE)
public @interface ApiActions {
    String CHECKIN = "Checkin";
    String WIPE = "Wipe";
    String LOCK = "Lock";
    String UNENROL = "Unenroll";
    String SEND_MESSAGE = "SendMessage";
    String LOCATE = "Locate";
    String SEND_SCRIPT = "SendScript";
    String ALLOW_EXCHANGE_ACCESS = "AllowExchangeAccess";
    String ALLOW_SOTI_SURF = "AllowSotiSurf";
    String APPLE_SOFTWARE_UPDATE_REFRESH_STATUS = "AppleSoftwareUpdateRefreshStatus";
    String APPLE_SOFTWARE_UPDATE_SCAN = "AppleSoftwareUpdateScan";
    String APPLE_SOFTWARE_UPDATE_SCHEDULE = "AppleSoftwareUpdateSchedule";
    String BLOCK_EXCHANGE_ACCESS = "BlockExchangeAccess";
    String BLOCK_SOTI_HUB = "BlockSotiHub";
    String BLOCK_SOTI_SURF = "BlockSotiSurf";
    String BYPASS_ACTIVATION_LOCK = "BypassActivationLock";
    String CLEAR_RESTRICTION = "ClearRestrictions";
    String CLEAR_SOTI_SURF_CACHE = "ClearSotiSurfCache";
    String DISABLE_AGENT_UPGRADE = "DisableAgentUpgrade";
    String DISABLE_LOST_MODE = "DisableLostMode";
    String DISABLE_PASSLOCK_CODE = "DisablePasscodeLock";
    String ENABLE_AGENT_UPGRADE = "EnableAgentUpgrade";
    String ENABLE_LOST_MODE = "EnableLostMode";
    String FACTORY_RESET = "FactoryReset";
    String MIMiGRATE_TO_ELM_AGENT = "grateToELMAgent";
    String RESET_PASSCODE = "ResetPasscode";
    String REMOTE_RING = "RemoteRing";
    String SCAN_FOR_VIRUS = "ScanForViruses";
    String FILE_SYNC_FILE_NOW = "SyncFilesNow";
    String SOFT_RESET = "SoftReset";
    String SEND_SCRIPT_VIA_SMS = "SendScriptViaSms";
    String SEND_TEST_PAGE = "SendTestPage";
    String TURN_OFF_SUSPEND = "TurnOffSuspend";
    String UPDATE_VIRUS_DEFINITION = "UpdateVirusDefinition";
    String UPGRADE_AGENT_NOW = "UpgradeAgentNow";
    String ENROLL_IN_EFOTA = "EnrollInEFOTA";
    String UPGRADE_FIRMWARE = "UpgradeFirmware";

}

File: app/src/main/java/com/mdmobile/cyclops/dataTypes/ApiActions.kt
package com.mdmobile.cyclops.dataTypes

import androidx.annotation.StringDef

/**
 * Constants to define possible api actions
 */
class ApiActions {
    @StringDef(CHECKIN, WIPE, LOCK, UNENROL, SEND_MESSAGE, LOCATE, SEND_SCRIPT, ALLOW_EXCHANGE_ACCESS, ALLOW_SOTI_SURF,
            APPLE_SOFTWARE_UPDATE_REFRESH_STATUS, APPLE_SOFTWARE_UPDATE_SCAN, APPLE_SOFTWARE_UPDATE_SCHEDULE, BLOCK_EXCHANGE_ACCESS)
    @kotlin.annotation.Retention(kotlin.annotation.AnnotationRetention.SOURCE)
    annotation class ApiActions

    companion object {
        const val CHECKIN = "Checkin"
        const val WIPE = "Wipe"
        const val LOCK = "Lock"
        const val UNENROL = "Unenroll"
        const val SEND_MESSAGE = "SendMessage"
        const val LOCATE = "Locate"
        const val SEND_SCRIPT = "SendScript"
        const val ALLOW_EXCHANGE_ACCESS = "AllowExchangeAccess"
        const val ALLOW_SOTI_SURF = "AllowSotiSurf"
        const val APPLE_SOFTWARE_UPDATE_REFRESH_STATUS = "AppleSoftwareUpdateRefreshStatus"
        const val APPLE_SOFTWARE_UPDATE_SCAN = "AppleSoftwareUpdateScan"
        const val APPLE_SOFTWARE_UPDATE_SCHEDULE = "AppleSoftwareUpdateSchedule"
        const val BLOCK_EXCHANGE_ACCESS = "BlockExchangeAccess"
        const val BLOCK_SOTI_HUB = "BlockSotiHub"
        const val BLOCK_SOTI_SURF = "BlockSotiSurf"
        const val BYPASS_ACTIVATION_LOCK = "BypassActivationLock"
        const val CLEAR_RESTRICTION = "ClearRestrictions"
        const val CLEAR_SOTI_SURF_CACHE = "ClearSotiSurfCache"
        const val DISABLE_AGENT_UPGRADE = "DisableAgentUpgrade"
        const val DISABLE_LOST_MODE = "DisableLostMode"
        const val DISABLE_PASSLOCK_CODE = "DisablePasscodeLock"
        const val ENABLE_AGENT_UPGRADE = "EnableAgentUpgrade"
        const val ENABLE_LOST_MODE = "EnableLostMode"
        const val FACTORY_RESET = "FactoryReset"
        const val MIMiGRATE_TO_ELM_AGENT = "grateToELMAgent"
        const val RESET_PASSCODE = "ResetPasscode"
        const val REMOTE_RING = "RemoteRing"
        const val SCAN_FOR_VIRUS = "ScanForViruses"
        const val FILE_SYNC_FILE_NOW = "SyncFilesNow"
        const val SOFT_RESET = "SoftReset"
        const val SEND_SCRIPT_VIA_SMS = "SendScriptViaSms"
        const val SEND_TEST_PAGE = "SendTestPage"
        const val TURN_OFF_SUSPEND = "TurnOffSuspend"
        const val UPDATE_VIRUS_DEFINITION = "UpdateVirusDefinition"
        const val UPGRADE_AGENT_NOW = "UpgradeAgentNow"
        const val ENROLL_IN_EFOTA = "EnrollInEFOTA"
        const val UPGRADE_FIRMWARE = "UpgradeFirmware"
    }
}


